docs/root/configuration/observability/access_log/usage.rst

```diff
@@ -456,6 +456,34 @@ The following command operators are supported:
     JSON struct or list is rendered. Structs and lists may be nested. In any event, the maximum
     length is ignored
 
+%CLUSTER_METADATA(NAMESPACE:KEY*):Z%
+  HTTP
+    :ref:`Upstream cluster Metadata <envoy_v3_api_msg_config.core.v3.Metadata>` info,
+    where NAMESPACE is the filter namespace used when setting the metadata, KEY is an optional
+    lookup up key in the namespace with the option of specifying nested keys separated by ':',
+    and Z is an optional parameter denoting string truncation up to Z characters long. The data
+    will be logged as a JSON string. For example, for the following dynamic metadata:
+
+    ``com.test.my_filter: {"test_key": "foo", "test_object": {"inner_key": "bar"}}``
+
+    * %CLUSTER_METADATA(com.test.my_filter)% will log: ``{"test_key": "foo", "test_object": {"inner_key": "bar"}}``
+    * %CLUSTER_METADATA(com.test.my_filter:test_key)% will log: ``"foo"``
+    * %CLUSTER_METADATA(com.test.my_filter:test_object)% will log: ``{"inner_key": "bar"}``
+    * %CLUSTER_METADATA(com.test.my_filter:test_object:inner_key)% will log: ``"bar"``
+    * %CLUSTER_METADATA(com.unknown_filter)% will log: ``-``
+    * %CLUSTER_METADATA(com.test.my_filter:unknown_key)% will log: ``-``
+    * %CLUSTER_METADATA(com.test.my_filter):25% will log (truncation at 25 characters): ``{"test_key": "foo", "test``
+
+  TCP
+    Not implemented ("-").
```

Comment:
I'm actually not sure that this is not implemented for TCP, I just copied it from the dynamic metadata documentation.
I can understand why HTTP filter setting metadata would not be implemented for TCP for the dynamic metadata.
For cluster metadata, however, I believe that the formatter has no issues with TCP, but I'm not sure that there is cluster metadata for TCP - @yanavlasov do you know how do that works?

Reply:
This metadata is for filters but I do not know if this applies to the TCP proxy as well. I will ask someone to look at this PR who might know.